### ISSUE TITLE
ValueError: The truth value of a DataFrame is ambiguous.

### ISSUE THREAD

**Issue by Zongshun96** [TrusteeML/emperor#5]
Greetings,
I am trying to reproduce the results for the provided use cases and facing some issues. Could you please have a look at where the problem is? Thank you.
BTW, the paper looks very nice to me. Great job!

When I reach ``python main.py`` in the IoT case, the code exited with the following error message. I also saw the same error in other use cases, i.e., heartbleed.

```
(venv) cc@expr-cuts-faas:~/study_trustee/emperor/use_cases/iot_case$ python main.py 
2023-01-29 21:47:27,301 - INFO - Reading dataset from CSV...
2023-01-29 21:47:28,681 - INFO - Done!
2023-01-29 21:47:28,681 - INFO - Running Trust Report...
2023-01-29 21:47:28,682 - INFO - Preparing data...
2023-01-29 21:47:28,682 - INFO - Splitting dataset for training and testing...
Traceback (most recent call last):
  File "main.py", line 57, in <module>
    main()
  File "main.py", line 38, in main
    trust_report = TrustReport(
  File "/home/cc/study_trustee/venv/lib/python3.8/site-packages/trustee/report/trust.py", line 156, in __init__
    self._prepare_data()
  File "/home/cc/study_trustee/venv/lib/python3.8/site-packages/trustee/report/trust.py", line 620, in _prepare_data
    self.y = pd.Series(self.y)
  File "/home/cc/study_trustee/venv/lib/python3.8/site-packages/pandas/core/series.py", line 231, in __init__
    if is_empty_data(data) and dtype is None:
  File "/home/cc/study_trustee/venv/lib/python3.8/site-packages/pandas/core/construction.py", line 596, in is_empty_data
    is_simple_empty = is_list_like_without_dtype and not data
  File "/home/cc/study_trustee/venv/lib/python3.8/site-packages/pandas/core/generic.py", line 1329, in __nonzero__
    raise ValueError(
ValueError: The truth value of a DataFrame is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().
```

Then, when I try with the docker method, the ``docker build . -t emperor-iot`` throws another error at ``RUN poetry config virtualenvs.create false && poetry install``.

```
(venv) cc@expr-cuts-faas:~/study_trustee/emperor/use_cases/iot_case$ sudo docker build . -t emperor-iot
Sending build context to Docker daemon  45.57kB
Step 1/13 : FROM ubuntu:19.10
 ---> 2f6c85efea61
Step 2/13 : ENV DEBIAN_FRONTEND=noninteractive
 ---> Using cache
 ---> 49e02bcbc9b9
Step 3/13 : SHELL ["/bin/bash", "-c"]
 ---> Using cache
 ---> 4dd89bf51794
Step 4/13 : RUN sed -i -e 's/archive.ubuntu.com\|security.ubuntu.com/old-releases.ubuntu.com/g' /etc/apt/sources.list
 ---> Using cache
 ---> 2936a3a9a1ab
Step 5/13 : RUN apt update && apt install -y libpcap-dev python3-dev python3-pip wget make build-essential libssl-dev zlib1g-dev   libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm curl git   libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev   graphviz fonts-roboto msttcorefonts
 ---> Using cache
 ---> f3877b1fb145
Step 6/13 : RUN rm ~/.cache/matplotlib -rf
 ---> Using cache
 ---> 58b414e40a84
Step 7/13 : RUN pip3 install --upgrade pip
 ---> Using cache
 ---> a6eef2dfcb72
Step 8/13 : RUN pip3 install poetry
 ---> Using cache
 ---> 73e118d5412b
Step 9/13 : RUN mkdir /emperor
 ---> Using cache
 ---> 3a3bb037291d
Step 10/13 : COPY . /emperor
 ---> Using cache
 ---> b27e4420ad7f
Step 11/13 : WORKDIR /emperor
 ---> Using cache
 ---> ce51ada57452
Step 12/13 : RUN pip3 install --ignore-installed PyYAML entrypoints
 ---> Using cache
 ---> 052f156ca46a
Step 13/13 : RUN poetry config virtualenvs.create false && poetry install
 ---> Running in b710a16f0abc
Skipping virtualenv creation, as specified in config file.

Invalid version '0.21ubuntu4' on package distro-info
The command '/bin/bash -c poetry config virtualenvs.create false && poetry install' returned a non-zero code: 1
```

Any help is appreciated. Thank you!

Best,
Zongshun

**Comment by asjacobs92** [comment]
Hey Zongshun! I'm glad you liked the paper!

So the error `ValueError: The truth value of a DataFrame is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().` happened a lot during development, when I tried to test if a Pandas data frame was null or not, but I thought I had cleared all of those issues. I'll take a look at the code to see if I can reproduce it. Can you check which version of Trustee you installed? 

About Docker related issue, it seems to be a distro issue with the packages in the lock file (since I generated them using a Macbook). Please try deleting the file and running the build command again.

**Comment by Chris-1201** [comment]
I have the same problem and i wonder know how to fix it. Thanks a lot!

**Comment by asjacobs92** [comment]
I found the issue that was causing this! Apparently Pandas doesn't like creating a Dataframe from another Dataframe. :)
Lesson learned.

All issues from the thread should be fixed now.
I updated the trustee package to version 1.1.4 in all use cases, and stopped using poetry to build the Docker containers.
See latest commit below.

https://github.com/TrusteeML/emperor/commit/4cd8a206fe0551803e057e612b13cb7ed1220cfe

I am closing the issue.
